src/main/scala/scala/concurrent/java8/FutureConvertersImpl.scala

```diff
@@ -66,8 +66,15 @@ object FuturesConvertersImpl {
       cf
     }
 
-    override def toCompletableFuture(): CompletableFuture[T] =
-      throw new UnsupportedOperationException("this CompletionStage represents a read-only Scala Future")
+    /**
+     * @inheritdoc
+     *
+     * WARNING: completing the result of this method will not complete the underlying
```

Comment:
What happens if the CF is completed by a Scala Future prior to `toCompletableFuture.complete(x)` gets called?